packages/flutter_tools/lib/src/project.dart

```diff
@@ -552,10 +556,54 @@ class AndroidProject extends FlutterProjectPlatform {
     return imperativeMatch || declarativeMatch;
   }
 
+  /// Gets top-level Gradle build file.
+  /// See https://developer.android.com/build#top-level.
+  ///
+  /// The file must exist and it must be written in either Groovy (build.gradle)
+  /// or Kotlin (build.gradle.kts).
+  File get hostAppGradleFile {
+    final File buildGroovy = hostAppGradleRoot.childFile('build.gradle');
+    final File buildKotlin = hostAppGradleRoot.childFile('build.gradle.kts');
+
+    if (buildGroovy.existsSync() && buildKotlin.existsSync()) {
```

Comment:
@johnmccutchan @christopherfujino  FYI new tool exit when a user has both kotlin and gradle build files.

Reply:
Nope, not anymore – too many test failures resulted that would be out of scope of this PR. I'll file an issue to keep track of it and link to that issue from the code comment.

Reply:
Sorry, I think I am *still* misunderstanding your comment. You are saying that after this change we DO tool exit if the user has both? Or DO NOT? (This line looks like we're silently preferring groovy.)

Reply:
I made some changes after I wrote the comment above. Sorry again for the confusion!

Currently, we DO NOT crash/exit the tool when both `build.gradle` and `build.gradle.kts` exist. Instead, we use `build.gradle` and ignore `build.gradle.kts` (and the same goes for `settings.gradle`). This is [consistent with Gradle's behavior](https://github.com/flutter/flutter/pull/140744#discussion_r1446440793).

Reply:
To clarify, when I said

> I think I am still misunderstanding your comment

I was referring to @reidbaker's first comment.